Helper/Cart.php

```diff
@@ -497,58 +497,67 @@ public function getCartData($payment_only, $place_order_payload, $quote = null)
+                $shipping_address = [
+                    'first_name' => $address->getFirstname(),
+                    'last_name' => $address->getLastname(),
+                    'company' => $address->getCompany(),
+                    'phone' => $address->getTelephone(),
+                    'street_address1' => $address->getStreetLine(1),
+                    'street_address2' => $address->getStreetLine(2),
+                    'locality' => $address->getCity(),
+                    'region' => $address->getRegion(),
+                    'postal_code' => $address->getPostcode(),
+                    'country_code' => $address->getCountryId(),
+                ];
 
-            $email = $shippingAddress->getEmail() ?: $email;
-            if ($email) {
-                $shipping_address['email'] = $email;
-            }
+                $email = $address->getEmail() ?: $email;
+                if ($email) {
+                    $shipping_address['email'] = $email;
+                }
 
-            foreach ($this->required_address_fields as $field) {
-                if (empty($shipping_address[$field])) {
-                    unset($shipping_address);
-                    break;
+                foreach ($this->required_address_fields as $field) {
+                    if (empty($shipping_address[$field])) {
+                        unset($shipping_address);
+                        break;
+                    }
                 }
-            }
 
-            if (@$shipping_address) {
-                $cost         = $shippingAddress->getShippingAmount();
-                $rounded_cost = $this->getRoundAmount($cost);
+                if (@$shipping_address) {
+                    $cost = $address->getShippingAmount();
+                    $rounded_cost = $this->getRoundAmount($cost);
 
-                $diff += $cost * 100 - $rounded_cost;
-                $totalAmount += $rounded_cost;
+                    $diff += $cost * 100 - $rounded_cost;
+                    $totalAmount += $rounded_cost;
 
-                $cart['shipments'] = [[
-                    'cost'             => $rounded_cost,
-                    'tax_amount'       => $this->getRoundAmount($shippingAddress->getShippingTaxAmount()),
-                    'shipping_address' => $shipping_address,
-                    'service'          => $shippingAddress->getShippingDescription(),
-                    'reference'        => $shippingAddress->getShippingMethod(),
-                ]];
+                    $cart['shipments'] = [[
+                        'cost' => $rounded_cost,
+                        'tax_amount' => $this->getRoundAmount($address->getShippingTaxAmount()),
+                        'shipping_address' => $shipping_address,
+                        'service' => $shippingAddress->getShippingDescription(),
+                        'reference' => $shippingAddress->getShippingMethod(),
+                    ]];
+                }
+            } else {
```

Comment:
nit: `if not A else B` is less readable. Let's change this to `if virtual then B else A`

Reply:
done